tests/unit/test_session.py

```diff
@@ -1624,6 +1624,216 @@ def test_stop_tuning_job_client_error(sagemaker_session):
     )
 
 
+def test_train_with_sagemaker_config_injection_custom_profiler_config(sagemaker_session):
```

Comment:
optional: might be able to parameterize these similar tests for cleaner code. Could reference some of the sagemaker_config util tests to see how to do parameterization like that

Reply:
Will optimize as a follow up in another PR